File: Problem-012/solution.py
def count(num):
  count = 0
  i= 1
  while i *i < num :
    if num %i ==0:
      count+= 1
      if i != num // i:
        count += 1
    i+=1
  return count 

def divisible_triangular_number(n):
  triangular =0 
  i = 1
  while true:
    triangle += 1
    if count(triangle) > n :
      return triangle
    i++
    

File: Problem-012/solution.js
function countDivisors(num) {
    let count = 0;

    for (let i = 1; i * i <= num; i++) {
        if (num % i === 0) {
            count++;

            if (i !== num / i) {
                count++;
            }
        }
    }

    return count;
}

function divisibleTriangleNumber(n) {
  let triangle = 0;
  let i = 1;
  while (true){
    triangle+=i;
    if (countDivisors(triangle)>n){
      return triangle;
    }
    i++;
  }
}

divisibleTriangleNumber(500);
